addons/mrp_subcontracting/models/stock_location.py

```diff
@@ -1,15 +1,108 @@
+                locations_per_company.setdefault(location.company_id, []).extend(location)
+        new_rules_vals = []
+        rules_to_unarchive = self.env['stock.rule']
+        for company, locations in locations_per_company.items():
+            reference_location_id = company.subcontracting_location_id
+            if reference_location_id:
+                reference_rules_from = self.env['stock.rule'].search([('location_src_id', '=', reference_location_id.id)])
+                reference_rules_to = self.env['stock.rule'].search([('location_dest_id', '=', reference_location_id.id)])
+                for location in locations:
+                    existing_rules = {
+                        (rule.route_id, rule.picking_type_id, rule.action, rule.location_src_id): rule
+                        for rule in self.env['stock.rule'].with_context(active_test=False).search([('location_src_id', '=', location.id)])
+                    }
+                    for rule in reference_rules_from:
+                        if (rule.route_id, rule.picking_type_id, rule.action, location) not in existing_rules:
+                            new_rules_vals.append(rule.copy_data({
+                                'location_src_id': location.id,
+                                'name': rule.name.replace(reference_location_id.name, location.name)
+                            })[0])
+                        else:
+                            existing_rule = existing_rules[(rule.route_id, rule.picking_type_id, rule.action, location)]
+                            if not existing_rule.active:
+                                rules_to_unarchive += existing_rule
+                    existing_rules = {
+                        (rule.route_id, rule.picking_type_id, rule.action, rule.location_dest_id): rule
+                        for rule in self.env['stock.rule'].with_context(active_test=False).search([('location_dest_id', '=', location.id)])
+                    }
+                    for rule in reference_rules_to:
+                        if (rule.route_id, rule.picking_type_id, rule.action, location) not in existing_rules:
+                            new_rules_vals.append(rule.copy_data({
+                                'location_dest_id': location.id,
+                                'name': rule.name.replace(reference_location_id.name, location.name)
+                            })[0])
+                        else:
+                            existing_rule = existing_rules[(rule.route_id, rule.picking_type_id, rule.action, location)]
+                            if not existing_rule.active:
+                                rules_to_unarchive += existing_rule
+        self.env['stock.rule'].create(new_rules_vals)
+        rules_to_unarchive.action_unarchive()
+
+    def _archive_subcontracting_location_rules(self):
+        """ Archive only subcontracting rules for the no more 'custom' subcontracting location(s)."""
```

Comment:
nitpick: English cleanup
`Archive subcontracting rules for locations that are no longer 'custom' subcontracting locations`